File: app/src/main/java/com/mediustechnologies/payemi/activities/act4BankList.java
package com.mediustechnologies.payemi.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.mediustechnologies.payemi.Models.getAllBanks;
import com.mediustechnologies.payemi.commons.urlconstants;
import com.mediustechnologies.payemi.commons.utils;
import com.mediustechnologies.payemi.helper.RetrofitClient;
import com.mediustechnologies.payemi.adapters.bankListAdapter;
import com.mediustechnologies.payemi.databinding.ActivityBankListBinding;
import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class act4BankList extends AppCompatActivity {

    private ActivityBankListBinding binding;
    private List<getAllBanks> banklist;
    private GridLayoutManager gridLayoutManager ;
    private bankListAdapter adapter;
    private final Context context = this;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityBankListBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        init();

        addItemsInRecyclerView();




    }


    private void addItemsInRecyclerView() {


        Log.d("tag","Access Token Saved in Utils "+utils.access_token);
        Call<List<getAllBanks>> call = RetrofitClient.getInstance(urlconstants.AuthURL).getApi().getAllBanks(utils.access_token);

        call.enqueue(new Callback<List<getAllBanks>>() {
            @Override
            public void onResponse(Call<List<getAllBanks>> call, Response<List<getAllBanks>> response) {

                if(response.code()==200) {
//                    Log.d("tag", "setdata: "+banklist.toString());
                    banklist  = response.body();
                    initRecyclerView();
                }
                else{
                    Log.d("tag", "Get BankList onResponse: "+response.code());
                }
            }

            @Override
            public void onFailure(Call<List<getAllBanks>> call, Throwable t) {
                Log.d("tag","Failed"+t.toString());
            }
        });




    }



    private void initRecyclerView() {
        RecyclerView bankRecyclerView = binding.listOfBanks;
        gridLayoutManager = new GridLayoutManager(context,3);
        gridLayoutManager.setOrientation(RecyclerView.VERTICAL);
        bankRecyclerView.setLayoutManager(gridLayoutManager);
        adapter = new bankListAdapter(banklist);
        bankRecyclerView.setAdapter(adapter);
        adapter.setOnItemClickListner(new bankListAdapter.onItemClicked() {
            @Override
            public void onItemClick(int position) {
                //on item click listner
                Intent i = new Intent(context,act5BankSubCategories.class);
                i.putExtra("name",banklist.get(position).getBank_name());
                i.putExtra("imgurl",banklist.get(position).getBank_logo_url());
                startActivity(i);
            }
        });




    }

    private void init(){

        binding.backButton.setOnClickListener(view -> finish());





    }
}

File: app/src/main/java/com/mediustechnologies/payemi/activities/act4BankList.kt
package com.mediustechnologies.payemi.activities

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import com.mediustechnologies.payemi.Models.bankListItem
import androidx.recyclerview.widget.GridLayoutManager
import com.mediustechnologies.payemi.adapters.bankListAdapter
import android.os.Bundle
import com.mediustechnologies.payemi.R
import androidx.recyclerview.widget.RecyclerView
import android.widget.Toast
import android.content.Intent
import android.view.View
import com.mediustechnologies.payemi.activities.act5BankSubCategories
import com.mediustechnologies.payemi.databinding.ActivityBankListBinding
import java.util.ArrayList

class act4BankList : AppCompatActivity() {
    private var binding: ActivityBankListBinding? = null
    private lateinit var banklist: MutableList<bankListItem>
    private var gridLayoutManager: GridLayoutManager? = null
    private var adapter: bankListAdapter? = null
    private val context: Context = this
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityBankListBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.root)
        init()
        addDummmyItemsInRecyclerView()
        initRecyclerView()
    }

    private fun addDummmyItemsInRecyclerView() {
        banklist = ArrayList()
        banklist.add(bankListItem(R.drawable.b1, "Axis Bank"))
        banklist.add(bankListItem(R.drawable.b2, "HDFC Bank"))
        banklist.add(bankListItem(R.drawable.b3, "SBI"))
        banklist.add(bankListItem(R.drawable.b4, "Kotak Bank"))
        banklist.add(bankListItem(R.drawable.b5, "Bank of Baroda"))
        banklist.add(bankListItem(R.drawable.b6, "City Bank"))
        banklist.add(bankListItem(R.drawable.b7, "HSBC Bank"))
        banklist.add(bankListItem(R.drawable.b8, "Edelweiss Bank"))
        banklist.add(bankListItem(R.drawable.b9, "ICICI Bank"))
    }

    private fun initRecyclerView() {
        val bankRecyclerView = binding!!.listOfBanks
        gridLayoutManager = GridLayoutManager(context, 3)
        gridLayoutManager!!.orientation = RecyclerView.VERTICAL
        bankRecyclerView.layoutManager = gridLayoutManager
        adapter = bankListAdapter(banklist)
        bankRecyclerView.adapter = adapter
        adapter!!.setOnItemClickListner { position ->
            Toast.makeText(
                context,
                "$position posion item clicked",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private fun init() {
        binding!!.backButton.setOnClickListener { view: View? -> finish() }
        binding!!.bharatBillLogo.setOnClickListener { view: View? ->
            startActivity(
                Intent(
                    this,
                    act5BankSubCategories::class.java
                )
            )
        }
    }
}